### ISSUE TITLE
Loss goes to zero - custom data

### ISSUE THREAD

**Issue by mohbattharani** [TIGER-AI-Lab/VLM2Vec#67]
<img width="736" alt="Image" src="https://github.com/user-attachments/assets/b162ff94-e4a9-4eab-a4c0-a4f669b786d3" />

I am training on my custom data of text-image pairs and the loss goes to zero. 

I looked at the `collator` code, it seems `neg samples` are never used by the current implementation: `collator.py [Line 175]`

```
@dataclass
class TrainTextImageDataCollator:
    data_args: DataArguments
    model_args: ModelArguments
    processor: ProcessorMixin

    def __call__(self, examples):
        """
        :param examples: qry, qry_image, pos_text, pos_image
        """

        qry_inputs = self._get_batch_inputs(examples, "query_text", "query_image")
        pos_inputs = self._get_batch_inputs(examples, "pos_text", "pos_image")
        neg_inputs = self._get_batch_inputs(examples, "neg_text", "neg_image")
        
        return qry_inputs, pos_inputs
```

**Comment by XMHZZ2018** [comment]
Hi @mohbattharani , "neg" refers to hard negatives in this context. In our current codebase, hard negatives are not used during model training because mining high-quality ones can be quite time-consuming. Instead, we leverage a large batch size with GradCache to incorporate more in-batch negatives. We plan to include hard negatives in the next version of VLM2Vec.

Regarding your question about the zero loss, it's unclear whether the model is overfitting to the training data. You might consider evaluating the model on a hold-out custom dataset to better assess its performance.

**Comment by mohbattharani** [comment]
I guess I was using smaller batch size = 8, increasing to 32 resolved the issue. Working on using loss with hard negatives for next step of the training. Thank you.
